### DISCUSSION TITLE
Adjusting references for external use

### DISCUSSION THREAD

**Discussion by flickerfly** [usnistgov/OSCAL#1057] [General]
I was poking around EasyDynamics' oscal-rest repo and found that they [forked OSCAL](https://github.com/EasyDynamics/OSCAL/blob/json-schema-ref-by-path/json/schema/oscal_catalog_schema.json
) so they could adjust the references in the JSON schema to be able to be pulled into their rest API. For example, they are able to reference the catalog schema like this:

`https://raw.githubusercontent.com/EasyDynamics/OSCAL/json-schema-ref-by-path/json/schema/oscal_catalog_schema.json#/definitions/assembly_oscal-catalog_catalog`

I tried adding the control schema directly from the NIST repos in the same way like below, but it failed when trying to pull in pointers to things like `assembly_oscal-catalog-common_parameter` as it couldn't find those.

`https://raw.githubusercontent.com/usnistgov/OSCAL/main/json/schema/oscal_complete_schema.json#/definitions/oscal-complete-oscal-catalog:control`

Using the EasyDynamics fork I could reference it similar to the catalog

`https://raw.githubusercontent.com/EasyDynamics/OSCAL/json-schema-ref-by-path/json/schema/oscal_catalog_schema.json#/definitions/assembly_oscal-catalog_control`

You can replicate what I did by adding this to the schema section using [Swagger Editor on the OpenAPI doc (Direct Link)](https://editor.swagger.io/?url=https://raw.githubusercontent.com/EasyDynamics/oscal-rest/develop/openapi.yaml).

```
    OSCALControl:
      $ref:
        "https://raw.githubusercontent.com/usnistgov/OSCAL/main/json/schema/oscal_complete_schema.json#/definitions/oscal-complete-oscal-catalog:control"
```

Anyway, after all that I'm wondering if there is a way to properly create these references without the sub-references breaking or forking OSCAL's repo to move to a workable reference structure. If not, is there a feature request that could be made or is it something that's already been discussed and won't be supported?

**Comment by flickerfly** [comment]
Found out the EasyDynamics is tracking this now in their issues:
https://github.com/EasyDynamics/oscal-rest/issues/32

**Reply by david-waltermire** [reply]
We have an issue for this: usnistgov/metaschema#160.

**Comment by rgauss** [comment]
@flickerfly, just to be clear, the `oscal-rest` issue you referenced is a future task to switch back to NIST's repo if/when the underlying issue is resolved.

The issue on the NIST side we've submitted is usnistgov/metaschema#160 which spawned from [comments on a related issue](https://github.com/usnistgov/OSCAL/issues/933#issuecomment-857604212).
